### PR TITLE
[Menu logo] From computer

### PR DESCRIPTION
**Pull request by ronronscelestes**
## What

- Initialization of upload modals stepper + modals reducer
- From computer upload with file's format and sizing front-end checks
- Included voice-over indications on file error

## Demo

- Go to Settings / Overview
- Open modal by clicking on "plus" icon button
- Try to upload (from browsing or drag&drop) a file with wrong formats and see error message
- Try to upload a file with right format and land on pending modal

- Use custom logo with temp/LogoAPI
- Open modal by clicking on "plus" icon button
- Land on pending modal

## Snapshots

image

image

image

### CODE REVIEW

**Comment by maevalienard** [comment]
One question here because I can't remember, do we have a min size? If that's not the case for the second type of error - the size - I'd go for something like "The file uploaded is too large (max dimension: 750*750px, max file size: TBC)"

**Review by soupette — APPROVED** [review]

File: packages/core/admin/admin/src/pages/SettingsPage/pages/ApplicationInfosPage/components/LogoInput/index.js [hunk 1]
```diff
@@ -1,58 +1,75 @@
...
 import { CarouselInput, CarouselSlide, CarouselActions } from '@strapi/design-system/CarouselInput';
 import { IconButton } from '@strapi/design-system/IconButton';
 import { Box } from '@strapi/design-system/Box';
 import Plus from '@strapi/icons/Plus';
+import LogoModalStepper from '../LogoModalStepper';
+import { SIZE, DIMENSION } from '../../utils/constants';
 
 const LogoInput = ({ customLogo, defaultLogo }) => {
   const { formatMessage } = useIntl();
+  const [isDialogOpen, setIsDialogOpen] = useState(true);
```
Comment (HichamELBSI) [inline packages/core/admin/admin/src/pages/SettingsPage/pages/ApplicationInfosPage/components/LogoInput/index.js:13]: Maybe we should rename this state to `isModalOpen`?

File: packages/core/admin/admin/src/pages/SettingsPage/pages/ApplicationInfosPage/temp/LogoAPI.js [hunk 2]
```diff
@@ -1,15 +1,15 @@
...
   constructor() {
-    this.menuLogo = {
-      url: CatLogo,
-      name: 'cat-logo.png',
-    };
-    // this.menuLogo = null
+    // this.menuLogo = {
+    //   url: CatLogo,
+    //   name: 'cat-logo.png',
+    // };
```
Comment (HichamELBSI) [inline packages/core/admin/admin/src/pages/SettingsPage/pages/ApplicationInfosPage/temp/LogoAPI.js:11]: Still need these lines ?

**Review by HichamELBSI — CHANGES_REQUESTED** [review]

File: packages/core/admin/admin/src/pages/SettingsPage/pages/ApplicationInfosPage/components/LogoInput/index.js (hunk 1 above)
Comment (ronronscelestes) [reply packages/core/admin/admin/src/pages/SettingsPage/pages/ApplicationInfosPage/components/LogoInput/index.js:13]: would make sense with the modal component name! thanks!

**Review by HichamELBSI — APPROVED** [review]
LGTM

**Pull request merged**